### PR TITLE
Fix ModelInfo cattrs serialization issue 

### PR DESCRIPTION
**Pull request by parano**
## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/_internal/models/model.py [hunk 1]
```diff
@@ -318,25 +328,30 @@ def __init__(self, base_path: "t.Union[PathType, FS]"):
...
 
     @staticmethod
-    def from_dict(data: dict[str, str | dict[str, str]]) -> ModelContext:
-        return cattrs.structure(data, ModelContext)
+    def from_dict(data: dict[str, str | dict[str, str]] | ModelContext) -> ModelContext:
+        if isinstance(data, ModelContext):
+            return data
+        return bentoml_cattr.structure(data, ModelContext)
+
+attr.resolve_types(ModelContext, globals(), locals())
```
Comment (parano) [inline bentoml/_internal/models/model.py:341]: @sauyon looks like this is required for cattrs to work, when using `from __future__ import annotations`, see https://www.attrs.org/en/stable/types.html

Comment (sauyon) [reply bentoml/_internal/models/model.py:341]: Ah, got it! I think according to the cattrs people `cattrs.Converter()` should also be able to structure these types.

Comment (sauyon) [reply bentoml/_internal/models/model.py:341]: Though probably for our types we want to do this anyway.

Comment (parano) [reply bentoml/_internal/models/model.py:341]: @sauyon cattrs.Convertor doesn't have a resolve_types method, it seems to be a attrs behavior that cattrs did not expect

**Comment by parano** [comment]
Besides using `attr.resolve_types`, attrs & cattrs doesn't seem to work with type annotation `dict`, had to change all the `attrs` class fields to use `t.Dict` instead. @sauyon

**Comment by sauyon** [comment]
> Besides using `attr.resolve_types`, attrs & cattrs doesn't seem to work with type annotation `dict`, had to change all the `attrs` class fields to use `t.Dict` instead. @sauyon 

Hmm, that's probably the best solution until we drop support for 3.7/3.8 😛

File: bentoml/_internal/bento/bento.py [hunk 2]
```diff
@@ -463,14 +474,18 @@ def validate(self):
...
 bentoml_cattr.register_unstructure_hook(
     BentoInfo,
     # Ignore tag, tag is saved via the name and version field
-    make_dict_unstructure_fn(BentoInfo, bentoml_cattr, tag=override(omit=True)),
+    make_dict_unstructure_fn(BentoInfo, bentoml_cattr, tag=override(omit=True)),  # type: ignore
 )
 
 
 def _BentoInfo_dumper(dumper: yaml.Dumper, info: BentoInfo) -> yaml.Node:
+    print(info.to_dict())
```
Comment (parano) [inline bentoml/_internal/bento/bento.py:488]:
```suggestion
```

**Pull request merged**